.brigade/brigade.ts

```diff
@@ -150,7 +164,7 @@ const buildJob = (event: Event, version?: string) => {
     env["IMAGE_REGISTRY_PASSWORD"] = registryPassword
     registriesLoginCmd = `${registriesLoginCmd} && docker login ${registry} -u ${registryUsername} -p $IMAGE_REGISTRY_PASSWORD`
   }
-  const job = new MakeTargetJob(["build"], dockerClientImg, event, env)
+  const job = new JobWithSource("build", dockerClientImg, event, env)
```

Comment:
This is the crux of the whole PR...

I don't like building a `MakeTargetJob` named `build` when the target actually invoked by this job is `push`. It's one thing for the build name and the target name not to match, but `MakeTargetJob` has always kinda carried with it this expectation that you just give it a target and it runs it and that's not what happens here. The behavior of this job is easier to understand if the job is based on a more generic `JobWithSource` class.